online/src/main/scala/ai/chronon/online/MetadataStore.scala

```diff
@@ -149,88 +148,31 @@ class MetadataStore(kvStore: KVStore, val dataset: String = ChrononMetadataKey,
       },
       { gb => Metrics.Context(environment = "group_by.serving_info.fetch", groupBy = gb) })
 
-  // upload the materialized JSONs to KV store:
-  // key = <conf_type>/<team>/<conf_name> in bytes e.g joins/team/team.example_join.v1 value = materialized json string in bytes
-  def putConf(configPath: String): Future[Seq[Boolean]] = {
-    val configFile = new File(configPath)
-    assert(configFile.exists(), s"$configFile does not exist")
-    logger.info(s"Uploading Chronon configs from $configPath")
-    val fileList = listFiles(configFile)
-
-    val puts = fileList
-      .filter { file =>
-        val name = parseName(file.getPath)
-        if (name.isEmpty) logger.info(s"Skipping invalid file ${file.getPath}")
-        name.isDefined
+  def put(
+      kVPairs: Map[String, Seq[String]],
+      datasetName: String = ChrononMetadataKey,
+      batchSize: Int = CONF_BATCH_SIZE
+  ): Future[Seq[Boolean]] = {
+    val puts = kVPairs.map {
+      case (k, v) => {
+        logger.info(s"""Putting metadata for
+             |dataset: $datasetName
+             |key: $k
+             |conf: $v""".stripMargin)
+        val kBytes = k.getBytes()
+        val vBytes =
+          if (v.isEmpty) Array.emptyByteArray
+          else if (v.length == 1) v.head.getBytes()
+          else v.mkString(",").getBytes()
```

Comment:
this will break if there are commas in a single member string of the seq.

Chatgpt gave me this

```scala
import java.util.Base64
import java.nio.charset.StandardCharsets

object StringArrayConverter {
  // Method to convert an array of strings to a byte array using Base64 encoding for each element
  def stringsToBytes(strings: Array[String]): Array[Byte] = {
    val base64EncodedStrings = strings.map(s => Base64.getEncoder.encodeToString(s.getBytes(StandardCharsets.UTF_8)))
    base64EncodedStrings.mkString("\n").getBytes(StandardCharsets.UTF_8)
  }

  // Method to convert a byte array back to an array of strings by decoding Base64
  def bytesToStrings(bytes: Array[Byte]): Array[String] = {
    val encodedString = new String(bytes, StandardCharsets.UTF_8)
    encodedString.split("\n").map(s => new String(Base64.getDecoder.decode(s), StandardCharsets.UTF_8))
  }
}

// Testing the methods
val originalStrings = Array("a", "b,c,d", "e.f.g", "h|i|j")
val bytes = StringArrayConverter.stringsToBytes(originalStrings)
val decodedStrings = StringArrayConverter.bytesToStrings(bytes)

println("Original Strings: " + originalStrings.mkString("[", ", ", "]"))
println("Decoded Strings: " + decodedStrings.mkString("[", ", ", "]"))
```

Reply:
use "," as you were using instead of the newline here

Reply:
done